explorer/src/components/account/NFTokenAccountHeader.tsx

```diff
@@ -0,0 +1,113 @@
+import { NftokenTypes } from "@glow-xyz/nftoken-js";
+import "bootstrap/dist/js/bootstrap.min.js";
+import React from "react";
+import { Account } from "../../providers/accounts";
+import {
+  parseNFTokenCollectionAccount,
+  parseNFTokenNFTAccount,
+} from "../../providers/accounts/utils/isNFTokenAccount";
+import { InfoTooltip } from "../common/InfoTooltip";
+import { CachedImageContent } from "../common/NFTArt";
+import { useNftokenMetadata } from "./nftoken-hooks";
+
+export function NFTokenAccountHeader({ account }: { account: Account }) {
+  const nft = parseNFTokenNFTAccount(account);
+
+  if (nft) {
+    return <NFTokenNFTHeader nft={nft} />;
+  }
+
+  const collection = parseNFTokenCollectionAccount(account);
+  if (collection) {
+    return <NFTokenCollectionHeader collection={collection} />;
+  }
+
+  return (
+    <>
+      <h6 className="header-pretitle">Details</h6>
+      <h2 className="header-title">Account</h2>
+    </>
+  );
+}
+
+export function NFTokenNFTHeader({ nft }: { nft: NftokenTypes.Nft }) {
+  const { data: metadata } = useNftokenMetadata(nft.metadata_url);
+
+  return (
+    <div className="row">
+      <div className="col-auto ms-2 d-flex align-items-center">
+        <CachedImageContent uri={metadata?.image} />
+      </div>
+
+      <div className="col mb-3 ms-0.5 mt-3">
+        {<h6 className="header-pretitle ms-1">NFToken NFT</h6>}
+        <div className="d-flex align-items-center">
+          <h2 className="header-title ms-1 align-items-center no-overflow-with-ellipsis">
+            {metadata ? metadata.name || "No NFT name was found" : "Loading..."}
+          </h2>
+        </div>
+
+        <div>
+          <div className={"d-inline-flex align-items-center mt-2"}>
+            <span className="badge badge-pill bg-dark">{`${
+              nft.authority_can_update ? "Mutable" : "Immutable"
+            }`}</span>
+
+            <InfoTooltip
+              bottom
+              text={
+                nft.authority_can_update
+                  ? "The authority of this NFT can update the Metadata."
+                  : "The Metadata cannot be updated by anyone."
+              }
+            />
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+export function NFTokenCollectionHeader({
+  collection,
+}: {
+  collection: NftokenTypes.Collection;
+}) {
+  const { data: metadata } = useNftokenMetadata(collection.metadata_url);
+
+  return (
+    <div className="row">
+      <div className="col-auto ms-2 d-flex align-items-center">
+        <CachedImageContent uri={metadata?.image} />
+      </div>
+
+      <div className="col mb-3 ms-0.5 mt-3">
+        {<h6 className="header-pretitle ms-1">NFToken Collection</h6>}
+        <div className="d-flex align-items-center">
+          <h2 className="header-title ms-1 align-items-center no-overflow-with-ellipsis">
+            {metadata
+              ? metadata.name || "No collection name was found"
+              : "Loading..."}
```

Comment:
When you rewrite your data fetcher to use React Suspense, you won't need to null check like this! See `useAnchorProgram`.